src/inc/clrnt.h

```diff
@@ -835,7 +835,7 @@ RtlVirtualUnwind_Unsafe(
 //  X86
 //
 
-#if defined(_TARGET_X86_)
+#if defined(_TARGET_X86_) && defined(WIN32)
```

Comment:
This ifdef block doesn't contain just the warning pragma, but also the _DISPATCHER_CONTEXT.
But thinking about it more, I think we should use !defined(FEATURE_PAL) instead, since the other place where the _DISPATCHER_CONTEXT is defined for x86 Linux is included for FEATURE_PAL only.
Btw, it seems the warning disabling pragmas are outdated, there is no function definition between the pragmas.